tests/infrastructure/golden_images/test_common_templates_data_volumes.py

```diff
@@ -35,22 +37,27 @@ def vm_from_golden_image_multi_storage(
 
 
 @pytest.fixture()
-def vm_from_golden_image(
+def fedora_vm_from_data_source(
     request,
     unprivileged_client,
+    golden_images_namespace,
     namespace,
-    ocs_storage_class,
-    golden_image_data_source_scope_function,
 ):
-    use_ocs_storage_class = request.param.get("ocs_storage_class")
-    storage_class = ocs_storage_class.name if use_ocs_storage_class else None
+    fedora_data_source = DataSource(client=unprivileged_client, name="fedora", namespace=golden_images_namespace.name)
+    storage_class = (
+        get_storage_class_from_matrix_different_from_data_source(data_source=fedora_data_source)
+        if request.param.get("set_storage_class")
+        else None
+    )
+    if request.param.get("set_storage_class") and storage_class is None:
+        pytest.xfail("storage_class_matrix only has 1 storage class defined")
     with VirtualMachineForTests(
-        name="vm-from-golden-image",
+        name="fedora-vm-from-data-source",
         namespace=namespace.name,
         client=unprivileged_client,
         memory_guest=Images.Fedora.DEFAULT_MEMORY_SIZE,
         data_volume_template=data_volume_template_with_source_ref_dict(
-            data_source=golden_image_data_source_scope_function, storage_class=storage_class
+            data_source=fedora_data_source, storage_class=storage_class
         ),
```

Comment:
maybe split this fixture, it does too many things. xfail can move to another fixture. wdyt